src/Profiles/Shifting/Shifter/ANTPLUS_ProfileShiftingShifter.cpp

```diff
@@ -0,0 +1,50 @@
+#include <Profiles/Shifting/Shifter/ANTPLUS_ProfileShiftingShifter.h>
+#include <CommonDataPages/ANTPLUS_CommonDataPagePrivateDefines.h>
+
+
+ProfileShiftingShifter::ProfileShiftingShifter(	uint16_t deviceNumber, uint8_t transmissionType) :
+    BaseMasterProfile(deviceNumber, transmissionType),
+    _patternStep( 0 ),
+    _toggle(0),
+    _eventCount( 0 )
+{
+    setChannelConfig();
+}
+
+void ProfileShiftingShifter::setChannelConfig() {
+    setChannelType(ANTPLUS_SHIFTING_MASTER_CHANNELTYPE);
+    setDeviceType(ANTPLUS_SHIFTING_DEVICETYPE);
+    setChannelPeriod(ANTPLUS_SHIFTING_CHANNELPERIOD);
+    setSearchTimeout(ANTPLUS_SHIFTING_SEARCHTIMEOUT);
+}
+
+void ProfileShiftingShifter::transmitNextDataPage() {
+
+    // some static aux messages (TODO preliminary solution)
+    const uint8_t manufacturer[]  = { 0x50, 0xFF, 0xFF, 0x01, 0x0F, 0x00, 0x85, 0x83 };
+    const uint8_t product[]       = { 0x51, 0xFF, 0xFF, 0x01, 0x01, 0x00, 0x00, 0x00 };
+
+    if (_patternStep++ < 64) {
+        transmitShiftingMainPageMsg();
+    }
+    else {
+        // TODO manufacturer and product handling
+        ShiftingBaseGenericMsg msg;
+        if (_toggle++ % 1 == 0)
+            msg.copyData((uint8_t*)manufacturer, MESSAGE_SIZE);
+        else
+            msg.copyData((uint8_t*)product, MESSAGE_SIZE);
+        send(msg);
+        _patternStep = 0;
+    }
+}
+
+void ProfileShiftingShifter::transmitShiftingMainPageMsg() {
+    ShiftingBaseMainDataPageMsg msg;
+    msg.setEventCount(_eventCount);
```

Comment:
Typically I avoid tracking any values in the profile to allow the users to track it easily how ever they want and reduce the memory footprint of the library. Also this way the user never has to guess whether a value is set by the library, the answer is "they always have to set it." I am welcome to suggestions but in my opinion uniformity is the safest way of not confusing endusers.

Also I don't see in the datasheet where eventCount increments on every 4 transmissions, I thought its only when the user shifts?

Reply:
I agree. Will change it.